text_generation/causal_lm/cpp/continuous_batching/library/src/tokenizer.cpp

```diff
@@ -29,12 +35,14 @@ class Tokenizer::Impl {
     }
 
     ov::Tensor encode(std::string prompt) {
+        std::unique_lock<std::mutex> lock(m_tokenizer_mutex);
```

Comment:
we also need to ensure that `ov::Tensor` is copied on caller side. 
E.g. we can return `ov::Tensor` while it's not copied yet, and override using next `encode` call

maybe we need to move copy from calling code to this place and always return cloned tensor?

Reply:
Yes, I think it would be safer to return a copy.